util/math_helper.h

```diff
@@ -45,6 +54,17 @@ int numElements(const cl::sycl::int4 &);
 int numElements(const cl::sycl::int8 &);
 int numElements(const cl::sycl::int16 &);
 
+template <typename T>
+int numElements(const cl::sycl::vec<T, 2> &) { return 2; }
```

Comment:
```cpp
template <typename T, int numElems>
int numElements(const cl::sycl::vec<T, numElems> &) { return numElems; }
```

This should work, right? And it should also replace all the `cl::sycl::float16` and similar overloads?